add documentation for LDAP security provider [CN-995] (#139)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/management-center-ldap.adoc

= LDAP Security Provider
:description: You can use your existing LDAP server for authentication/authorization on the Management Center.

{description}

See xref:management-center:deploy-manage:ldap.adoc[Management Center Documentation] for more detailed information.

== Setting Up the LDAP Security Provider

To set up LDAP, you need to configure the `securityProvider.ldap` section in the Management Center CR.

> The following are the required field to configure the LDAP Security provider.

Reviewer: ```suggestion
The following are the required fields to configure the LDAP Security provider.
```

[cols="30%m,70%a"]
|===
|Field|Description

|`credentialsSecretName`
| The name of the secret that contains `username` and `password` keys of a user that has admin privileges on the LDAP server. The `username` must be the DN of the user. It is used to connect to the server when authenticating users.

> 

Reviewer: Maybe it is better to show the expanded word for DN when it is used for the first time. Something like:

"... must be the distinguished name (DN) of the user. ..."

|`userDn`
| DN to be used for searching users.

|`groupDn`
| DN to be used for searching groups.

|`adminGroups`
| Members of these groups and its nested groups have admin privileges on the Management Center.

|`userGroups`
| Members of these groups and its nested groups have read and write privileges on the Management Center.

|`readonlyUserGroups`
| Members of these groups and its nested groups have only read privilege on the Management Center.

|`metricsOnlyGroups`
| Members of these groups and its nested groups have the privilege to see only the metrics on the Management Center.

|`userSearchFilter`
| LDAP search filter expression to search for the users.

|`groupSearchFilter`
| LDAP search filter expression to search for the groups.

|===

> === Example Management Center LDAP configuration

Reviewer: ```suggestion
=== Example Management Center LDAP Configuration
```

> Bellow is the example configuration for the LDAP Security Provider:

Reviewer: ```suggestion
The following is an example configuration for the LDAP Security Provider:
```

[source,yaml,subs="attributes+"]
----
include::ROOT:example$/management-center-ldap.yaml[]
----

> And the exmaple of how to create a `Secret` for the LDAP credentials:

Reviewer: ```suggestion
The following example shows how to create a `Secret` for the LDAP credentials.
```
